articles/quickstart/webapp/golang/interactive.md

```diff
@@ -0,0 +1,172 @@
+and return our routes using [github.com/gin-gonic/gin](https://github.com/gin-gonic/gin). You will be passing an
+instance of `Authenticator` to the method, so it can be used within the `login` and `callback` handlers.
+
+::: note
+The router uses the [github.com/gin-contrib/sessions](https://github.com/gin-contrib/sessions) middleware to manage our cookie based sessions.
+:::
+
+## Serve your application {{{ data-action=code data-code="main.go" }}}
+
+With both the authenticator and router configured, we can wire things up using our
+application's entry point. Inside `main.go`, create an instance of the authenticator and the router, which gets passed the authenticator instance.
+
+If you are using a `.env` file, like we are in this example, be sure to call `godotenv.Load()` at the very beginning of the `main()` function.
+
+## Add login to your application {{{ data-action=code data-code="login.go" }}}
+
+In order for user to be able to authenticate themselves, we need to create a handler function to handle the `/login` route.
+
+Create a file called `login.go` in the `web/app/login` folder, and add a `Handler` function. Upon executing the handler, the user will be redirect to Auth0 where they can enter their credentials.
+
+In order to be able to call the `/login` route, add a link to `/login` in the `home.html` template located in the `web/template` directory.
+
+```html
+<!-- web/template/home.html -->
+<div>
+    <h3>Auth0 Example</h3>
+    <p>Zero friction identity infrastructure, built for developers</p>
+    <a href="/login">SignIn</a>
+</div>
+```
+
+## Handle authentication callback {{{ data-action=code data-code="callback.go" }}}
+
+Once users have authenticated using Auth0's Universal Login Page, they'll return to the app at the `/callback` route.
+
+Create a file called `callback.go` in the `web/app/callback` folder, and add a `Handler` function.
+
+This handler will take the `code` querystring, provided by Auth0, and exchange it for an Id Token and an Access Token.
+
+If the Id token is valid, it will store the profile information (which gets build based on the claims in the Id token) in the session along with the access token, this is done in order to be able to retrieve and use that information when needed.
+
+## Display user profile information {{{ data-action=code data-code="user.go" }}}
+
+Now that your users can log in, you will likely want to be able to retrieve and use the [profile information](/users/concepts/overview-user-profile) associated with authenticated users. 
+
+You can access that [profile information](/users/concepts/overview-user-profile), such as their nickname or profile picture, through the `profile` that was stored in the session previously.
+
+Create a handler for the `/user` endpoint in `web/app/user/user.go` and return the corresponding HTML file. As the `profile` if being passed to `ctx.HTML()`, you can access the profile information such as `picture` and `nickname` inside that same HTML file. 
```

Comment:
```suggestion
Create a handler for the `/user` endpoint in `web/app/user/user.go` and return the corresponding HTML file. As the `profile` passes to `ctx.HTML()`, you can access the profile information such as `picture` and `nickname` inside that same HTML file. 
```

Reply:
Wouldn't `passes` be the incorrect tense here? I see it was also wrong in the original, it was supposed to be `is being passed`.